src/controllers/module/event/eventController.ts

```diff
@@ -123,4 +122,53 @@ export class EventController {
 			response.render('page/course/module/events/events-preview')
 		}
 	}
+
+	public getLocation() {
+		return async (req: Request, res: Response) => {
+			res.render('page/course/module/events/event-location', {event: req.session!.event})
+		}
+	}
+
+	public setLocation() {
+		return async (req: Request, res: Response) => {
+			const data = {venue: {
+					location: req.body.location,
+					address: req.body.address,
+					capacity: parseInt(req.body.capacity, 10),
+					minCapacity: parseInt(req.body.minCapacity, 10),
+				}}
+
+ 			const errors = await this.eventValidator.check(data, ['event.location'])
+			const event = await this.eventFactory.create(data)
+
+			if (errors.size) {
+				req.session!.sessionFlash = {errors: errors}
+				req.session!.event = req.session!.event ? function() {
+					const mergedEvent = req.session!.event
+					mergedEvent.venue = event.venue
+					return mergedEvent
+				}() : event
+				req.session!.save(() => {
+					res.redirect(`/content-management/courses/${req.params.courseId}/modules/${req.params.moduleId}/events/location`)
+				})
+			} else {
```

Comment:
You shouldnt need an `else` here

Reply:
mind you I've removed `return`s there, so w/o `else` it will keep going executing the latter block